src/vi/goctruyentranhvui/src/eu/kanade/tachiyomi/extension/vi/goctruyentranhvui/GocTruyenTranhVui.kt

```diff
@@ -0,0 +1,181 @@
+            .asObservableSuccess()
+            .map { response ->
+                val res = response.parseAs<ChapterDTO>()
+                res.result.chapters.map { itm ->
+                    SChapter.create().apply {
+                        name = itm.numberChapter
+                        date_upload = dateFormat.tryParse(itm.stringUpdateTime)
+                        setUrlWithoutDomain("$baseUrl${manga.url}/chuong-${itm.numberChapter}")
+                    }
+                }
+            }
+    }
+
+    private fun checkChapterLists(document: Document) = document.selectFirst("input[id=comic-id-comment]")!!.attr("value")
+
+    override fun chapterFromElement(element: Element): SChapter = throw UnsupportedOperationException()
+
+    override fun chapterListSelector(): String = ""
+
+    override fun imageUrlParse(document: Document): String = ""
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        element.selectFirst("a.mt-1").let {
+            setUrlWithoutDomain(it!!.absUrl("href"))
+            title = it.text()
+        }
+        thumbnail_url = element.selectFirst("img.lazy")!!.absUrl("data-original")
+    }
+
+    override fun latestUpdatesNextPageSelector(): String = ".ma-3 a"
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(
+        baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegments("truyen-cap-nhat")
+            addQueryParameter("p", page.toString())
+        }.build(),
+        headers,
+    )
+
+    override fun latestUpdatesSelector(): String = ".row .col-md-2"
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.select(".v-card-title").text()
+        genre = document.select(".group-content > .v-chip-link").joinToString { it.text().trim(',', ' ') }
+        thumbnail_url = document.selectFirst("img.image")!!.absUrl("src")
+        status = parseStatus(document.select(".mb-1:contains(Trạng thái:) span").text())
+        author = document.select(".mb-1:contains(Tác giả:) span").text()
+        description = document.select(".v-card-text").text()
+    }
+
+    private fun parseStatus(status: String?) = when {
+        status == null -> SManga.UNKNOWN
+        status.contains("Đang thực hiện", ignoreCase = true) -> SManga.ONGOING
+        status.contains("Hoàn thành", ignoreCase = true) -> SManga.COMPLETED
+        else -> SManga.UNKNOWN
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val html = response.body.string()
+        val pattern = Regex("chapterJson:\\s*`(.*?)`", RegexOption.DOT_MATCHES_ALL)
+        val match = pattern.find(html)
+        val json = Json {
+            ignoreUnknownKeys = true
+        }
+        val jsonString = match?.groups?.get(1)?.value ?: error("Không tìm thấy chapterJson")
+        val result = json.decodeFromString<ChapterWrapper>(jsonString)
```

Comment:
Use the `parseAs()` method from the core library to reuse the default `Json` instance.

Reply:
> Use the `parseAs()` method from the core library to reuse the default `Json` instance.

In this case, using `parseAs()` is not appropriate because I get the input data as string.

Reply:
The `parseAs()` has an overload on strings. Just type it in your IDE and you'll see it.

Even if you couldn't use `parseAs()`, you should use the default `jsonInstance` instead of creating one without special settings. The default instance already has `ignoreUnknownKeys = true`.